### PR TITLE
Fixed a truncation bug in the NUnit profile

### PR DESCRIPTION
**Pull request by kohsuke**
Turns out there's another odd behaviour with NUnitXML.Logger that we run into with a customer, which involves the use of TestCase annotation with parameter.

Improved the logic, and added a test case.

### CODE REVIEW

File: launchable/test_runners/nunit.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def build_path(e: Element):
         # work around a bug in NUnitXML.Logger.
         # see nunit-reporter-bug-with-nested-type.xml test case
         methodname = e.attrs['methodname']
-        idx = methodname.rfind(".")
+        bra = methodname.find("(")
+        idx = methodname.rfind(".",0,bra)
```
Comment (Konboi) [inline launchable/test_runners/nunit.py:48]: FYI: https://github.com/launchableinc/cli/actions/runs/5815538212/job/15767255587?pr=614#step:9:21

File: tests/data/nunit/nunit-reporter-bug-with-nested-type.json [hunk 2]
```diff
@@ -16,7 +16,25 @@
...
+    },
+    {
+      "type": "case",
+      "testPath": [
+        { "type": "Assembly",    "name": "Launchable.NUnit.Test.dll" },
+        { "type": "TestSuite",   "name": "Launchable" },
+        { "type": "TestSuite",   "name": "NUnit" },
+        { "type": "TestSuite",   "name": "Test" },
+        { "type": "TestSuite",   "name": "InlineParameterTests" },
+        { "type": "TestCase",    "name": "StrTest(\"dot.in.it\")" }
```
Comment (Konboi) [inline tests/data/nunit/nunit-reporter-bug-with-nested-type.json:28]: note:  `methodname="StrTest(&quot;dot.in.it&quot;)"`

**Review by Konboi — APPROVED** [review]
LGTM

Please merge after fixing lint errors.

**Comment by kohsuke** [comment]
Oops, I somehow always forget that the linter is in use here. It looks like I'm supposed to use pre-commit so let me fix that as well...

File: tests/data/nunit/nunit-reporter-bug-with-nested-type.json (hunk 2 above)
Comment (kohsuke) [reply tests/data/nunit/nunit-reporter-bug-with-nested-type.json:28]: Are you asking why this is the correct assertion?

Comment (Konboi) [reply tests/data/nunit/nunit-reporter-bug-with-nested-type.json:28]: No, just for my confirmation. Sorry if I made you confused.

**Pull request merged**